pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -198,12 +198,112 @@ func (c *Cluster) validateOSDSettings() error {
 	return nil
 }
 
+func (c *Cluster) ValidateTopologyAcrossNodes() error {
+	nodes, err := k8sutil.GetKubernetesNodesMatchingRookNodes(c.clusterInfo.Context, c.spec.Storage.Nodes, c.context.Clientset)
+	if err != nil {
+		return errors.Wrap(err, "failed to list nodes for topology validation")
+	}
+
+	return checkTopologyConflicts(nodes)
```

Comment:
Checking the topology is a somewhat expensive operation, especially for larger clusters that could have hundreds of nodes. The topology is unlikely to change, so it's not really necessary to analyze the topology with every reconcile. I'm thinking that validating successfully once each time the operator starts will be acceptable. This could be implemented with a bool package variable that is set after the successful topology check.

Reply:
done